### ISSUE TITLE
no-descending-specificity with "empty" definition

### ISSUE THREAD

**Issue by bkdotcom** [stylelint/stylelint#4637]
# Clearly describe the bug

"false-positive" with "empty" definitions  (see below)

# Which rule, if any, is the bug related to?

no-descending-specificity

# What code is needed to reproduce the bug?

```css
    nav {
      display: inline-block;
      a {
        display: inline-block;
      }
    }

    a,
    button,
    label {
      &:focus,
      &:hover {
        color: #333;
        text-decoration: none;
      }
    }
```

results in 
`Expected selector "a" to come before selector "nav a"`
(there is no "a" selector... only a:focus & a:hover)!

swapping to 
```css
    a,
    button,
    label {
      &:focus,
      &:hover {
        color: #333;
        text-decoration: none;
      }
    }

    nav {
      display: inline-block;
      a {
        display: inline-block;
      }
    }
```
results in 
`Expected selector "nav a" to come before selector "a:focus"`

# What stylelint configuration is needed to reproduce the bug?

"default"

```json
{
  "extends": "stylelint-config-standard",
  "rules": {
    "rule-empty-line-before": null,
	"at-rule-no-unknown": [ true, {
      "ignoreAtRules": ["extend", "at-root", "debug", "warn", "error", "if", "else", "for", "each", "while", "mixin", "include", "content", "return", "function"]
    }]
  }
}
```

# Which version of stylelint are you using?

`13.2.0`

# How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `npx stylelint ./scss`

# Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

¯\\\_(ツ)_/¯
Yes, it's related to SCSS nested properties?

# What did you expect to happen?

No warnings to be flagged.

# What actually happened (e.g. what warnings or errors did you get)?

```shell
Expected selector "a" to come before selector "nav a"`
```
(there is no "a" selector... only a:focus & a:hover)!

**Comment by jeddy3** [comment]
@bkdotcom Thanks for recreating the issue using the template.

I think this is a bug as the rule shouldn't check the selectors of rule-sets without declaration blocks.

I suspect the fix will require tweaking where [we resolve the nested selectors](https://github.com/stylelint/stylelint/blob/343e40dcb625e4d1001db441e4dcf7ac105784cf/lib/rules/no-descending-specificity/index.js#L76-L85).

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by JounQin** [comment]
It should have been fixed, [playground](https://stylelint.io/demo#N4Igxg9gJgpiBcIAEKkDsCGA3JwA6aqqUAlgM4AOANhgJ7xIlpVMwC0ARlRGANYDcBIigy4hwlKUo16jZq07c+gwsIC+4lBrSakGADS6OAVwAupiGkOqUNDjCpibqAGTwAZj2NlrEtwAsILBgAJycJFEhuEIYAYgBmRJUIpFMYAA9TNlhIEIxTEksGNEsYZPVdNRB9EDJaNFMMdIRwMjJq8Et3EgBzFvxVPBAQ4yoYMiGGAaIhkuzjahIwfPYyBxgwCxCJhFSRmHFtKrUgA)

**Comment by ybiquitous** [comment]
@JounQin Is this issue for `no-descending-specificity`, not `no-duplicate-selectors`?

**Comment by JounQin** [comment]
> @JounQin Is this issue for `no-descending-specificity`, not `no-duplicate-selectors`?

Ah, my bad.

**Comment by ybiquitous** [comment]
Does this false positive occur still? It seems the rule works on the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGAaAOgOwEYFcAueEG6GANkljKQATCbXUBk8AZhGDgM4kPMAWEAG4wATrXoNqkUhBHxqAYgDMKgNwSGeGAA88AWliQRSPAEsi8jERjqMDAL6ZHGTBiSDxd6lFOcADuQAnvKmZKEwelgyYADWtgxInpLevgFIwdShpOGR0XESzs4gKCCcgRh4SNoIpWCcnMXgRCymAOY1dF5oICI4pDCc3fKdyd1WBgNgMBg+GK16-jBgpi3LeIFD1Hi9MAVOIPZAA).

**Comment by jeddy3** [comment]
Let's close.
